src/pendulum/__init__.py

```diff
@@ -292,11 +292,7 @@ def from_timestamp(timestamp: int | float, tz: str | Timezone = UTC) -> DateTime
     """
     Create a DateTime instance from a timestamp.
     """
-    dt = _datetime.datetime.utcfromtimestamp(timestamp)
-
-    dt = datetime(
-        dt.year, dt.month, dt.day, dt.hour, dt.minute, dt.second, dt.microsecond
-    )
+    dt = DateTime.fromtimestamp(timestamp, tz=UTC)
```

Comment:
I think the current suggestion is recursive and does not use the correct keyword:
```suggestion
    dt = _datetime.datetime.fromtimestamp(timestamp, tzinfo=UTC)
```

For the develop branch, the change must be done in `pendulum/pendulum.py:491`
```python
dt = datetime.datetime.utcfromtimestamp(timestamp).replace(tzinfo=UTC)
```
to
```python
dt = datetime.datetime.fromtimestamp(timestamp, tzinfo=UTC)
```